nixos/modules/services/networking/kasmvnc.nix

```diff
@@ -0,0 +1,641 @@
+          rate_limit = cfg.data_loss_prevention.clipboard.keyboard.rate_limit;
+        };
+        logging = {
+          level = cfg.data_loss_prevention.clipboard.logging.level;
+        };
+      };
+    };
+    encoding = {
+      max_frame_rate = cfg.encoding.max_frame_rate;
+      full_frame_updates = cfg.encoding.full_frame_updates;
+      rect_encoding_mode = {
+        min_quality = cfg.encoding.rect_encoding_mode.min_quality;
+        max_quality = cfg.encoding.rect_encoding_mode.max_quality;
+        consider_lossless_quality = cfg.encoding.rect_encoding_mode.consider_lossless_quality;
+        rectangle_compress_threads = cfg.encoding.rect_encoding_mode.rectangle_compress_threads;
+      };
+      video_encoding_mode = {
+        jpeg_quality = cfg.encoding.video_encoding_mode.jpeg_quality;
+        webp_quality = cfg.encoding.video_encoding_mode.webp_quality;
+        max_resolution = {
+          width = cfg.encoding.video_encoding_mode.max_resolution.width;
+          height = cfg.encoding.video_encoding_mode.max_resolution.height;
+        };
+        enter_video_encoding_mode = {
+          time_threshold = cfg.encoding.video_encoding_mode.enter_video_encoding_mode.time_threshold;
+          area_threshold = cfg.encoding.video_encoding_mode.enter_video_encoding_mode.area_threshold;
+        };
+        exit_video_encoding_mode = {
+          time_threshold = cfg.encoding.video_encoding_mode.exit_video_encoding_mode.time_threshold;
+        };
+        logging = {
+          level = cfg.encoding.video_encoding_mode.logging.level;
+        };
+        scaling_algorithm = cfg.encoding.video_encoding_mode.scaling_algorithm;
+      };
+      compare_framebuffer = cfg.encoding.compare_framebuffer;
+      zrle_zlib_level = cfg.encoding.zrle_zlib_level;
+      hextile_improved_compression = cfg.encoding.hextile_improved_compression;
+    };
+    server = {
+      http = {
+        headers = cfg.server.http.headers;
+        httpd_directory = cfg.server.http.httpd_directory;
+      };
+      advanced = {
+        x_font_path = cfg.server.advanced.x_font_path;
+        kasm_password_file = lib.mkIf cfg.server.advanced.kasm_password_file cfg.server.advanced.kasm_password_file;
+        x_authority_file = cfg.server.advanced.x_authority_file;
+      };
+      auto_shutdown = {
+        no_user_session_timeout = cfg.server.auto_shutdown.no_user_session_timeout;
+        active_user_session_timeout = cfg.server.auto_shutdown.active_user_session_timeout;
+        inactive_user_session_timeout = cfg.server.auto_shutdown.inactive_user_session_timeout;
+      };
+    };
+    command_line = {
+      prompt = cfg.command_line.prompt;
+    };
+  };
+
+  format = pkgs.formats.yaml { };
+  configFileName = "/etc/kasmvnc/kasmvnc.yaml";
+  configFile = format.generate configFileName kasmvncConfig;
+
+in
+{
+  options = {
+    services.kasmvnc = {
+      enable = lib.mkEnableOption "KasmVNC service";
+
+      desktop = {
```

Comment:
This looks like a PITA to write and maintain and I don't see a benefit in replicating every upstream option; it's a fool's errand.

Please just make this an [RFC42-style settings option](https://github.com/NixOS/rfcs/blob/master/rfcs/0042-config-option.md).

What you can do is provide some abstract options that set the relevant settings internally if that even makes sense to do.